test/gradient/elixir_expr_test.exs

```diff
@@ -0,0 +1,373 @@
+              v
+
+            {:error, _} ->
+              :error
+          end
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "fn {:ok, v} -> v; {:error, _} -> :error end" == actual
+    end
+
+    test "binary comprehension" do
+      actual =
+        elixir_to_ast do
+          pixels = <<213, 45, 132, 64, 76, 32, 76, 0, 0, 234, 32, 15>>
+          for <<r::8, g::8, b::8 <- pixels>>, do: {r, g, b}
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "pixels = <<213, 45, 132, 64, 76, 32, 76, 0, 0, 234, 32, 15>>; for <<r::8, g::8, b::8 <- pixels >>, do: {r, g, b}" ==
+               actual
+    end
+
+    test "binary comprehension 2" do
+      actual =
+        elixir_to_ast do
+          for <<one, (_rest::binary-size(3) <- <<1, 2, 3, 4>>)>>, do: one
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "for <<one, _rest::binary-size(3) <- <<1, 2, 3, 4>> >>, do: one" == actual
+    end
+
+    test "receive" do
+      actual =
+        elixir_to_ast do
+          receive do
+            {:hello, msg} -> msg
+          end
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "receive do {:hello, msg} -> msg end" == actual
+    end
+
+    test "receive after" do
+      actual =
+        elixir_to_ast do
+          receive do
+            {:hello, msg} -> msg
+          after
+            1_000 -> "nothing happened"
+          end
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "receive do {:hello, msg} -> msg after 1000 -> \"nothing happened\" end" == actual
+    end
+
+    test "call pipe" do
+      actual =
+        elixir_to_ast do
+          [1, 2, 3]
+          |> Enum.map(fn x -> x + 1 end)
+          |> Enum.map(&(&1 + 1))
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "Enum.map(Enum.map([1, 2, 3], fn x -> x + 1 end), fn _ -> _ + 1 end)" == actual
+    end
+
+    test "with" do
+      actual =
+        elixir_to_ast do
+          map = %{a: 12, b: 0}
+
+          with {:ok, a} <- Map.fetch(map, :a),
+               {:ok, b} <- Map.fetch(map, :b) do
+            a + b
+          else
+            :error ->
+              0
+          end
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "map = %{a: 12, b: 0}; case :maps.find(:a, map) do {:ok, a} -> case :maps.find(:b, map) do {:ok, b} -> a + b; _gen -> case _gen do :error -> 0; _gen -> raise {:with_clause, _gen} end end; _gen -> case _gen do :error -> 0; _gen -> raise {:with_clause, _gen} end end" ==
+               actual
+    end
+
+    test "try reraise" do
+      actual =
+        elixir_to_ast do
+          try do
+            raise "ok"
+          rescue
+            e ->
+              IO.puts(Exception.format(:error, e, __STACKTRACE__))
+              reraise e, __STACKTRACE__
+          end
+        end
+        |> ElixirExpr.pp_expr()
+
+      assert "try do raise \"ok\"; catch :error, e -> IO.puts(Exception.format(:error, e, __STACKTRACE__)); reraise e, __STACKTRACE__ end" ==
```

Comment:
We could also use [sigils](https://elixir-lang.org/getting-started/sigils.html#strings) to avoid escaping nested quotation marks.